### PR TITLE
Module name changes structure change

### PR DESCRIPTION
**Pull request by zhumin8**
This is part 1 of implementation for #1383.
Changes include:
- Change folder name for generator scripts to 'spring-cloud-generator'
    - generation script changes to adapt to new folder names.
    - `generate-one.sh` change to add generated module to `spring-cloud-gcp-starters`
- Generated modules target destination renamed to `spring-cloud-previews` folder.
    - Changes to pom.xml to make it an aggregator for generated modules.
- Moved `fmt-maven-plugin` to spring-cloud-gcp/pom with default set to skip.
- Fix `spring-cloud-gcp-dependencies` version inheritance issue.
- `spring-cloud-gcp-starters/pom.xml` format change: replaced tabs with whitespaces, added comment marker for inserting generated modules.

Changes that needs to happen on the generator side:
- generated module pom changes: https://github.com/googleapis/gapic-generator-java/pull/1141
    - use `spring-cloud-gcp-starters` as parent
    - Added `.preview` suffix to generated module versions.

### CODE REVIEW

File: spring-cloud-gcp-starters/pom.xml [hunk 1]
```diff
@@ -1,47 +1,49 @@
...
-		<module>spring-cloud-gcp-starter-sql-postgres-r2dbc</module>
-		<module>spring-cloud-gcp-starter-security-iap</module>
-		<module>spring-cloud-gcp-starter-vision</module>
-		<module>spring-cloud-gcp-starter-data-firestore</module>
-		<module>spring-cloud-gcp-starter-bigquery</module>
-		<module>spring-cloud-gcp-starter-security-firebase</module>
-		<module>spring-cloud-gcp-starter-secretmanager</module>
-		<module>spring-cloud-gcp-starter-kms</module>
-	</modules>
+    xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/pom.xml:3]: Can we do the file reformat separately to aid reviewing?

File: spring-cloud-gcp-starters/pom.xml [hunk 2]
```diff
@@ -1,47 +1,49 @@
...
+    <module>spring-cloud-gcp-starter-sql-postgres-r2dbc</module>
+    <module>spring-cloud-gcp-starter-security-iap</module>
+    <module>spring-cloud-gcp-starter-vision</module>
+    <module>spring-cloud-gcp-starter-data-firestore</module>
+    <module>spring-cloud-gcp-starter-bigquery</module>
+    <module>spring-cloud-gcp-starter-security-firebase</module>
+    <module>spring-cloud-gcp-starter-secretmanager</module>
+    <module>spring-cloud-gcp-starter-kms</module>
+
+    <!--  preview modules  -->
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/pom.xml:47]: So, is the list of generated modules going here or into `spring-cloud-previews/pom.xml`?

Comment (zhumin8) [reply spring-cloud-gcp-starters/pom.xml:47]: Yes, the generated modules will be added here. example [here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/draft-generated/spring-cloud-gcp-starters/pom.xml#L47)

Comment (zhumin8) [reply spring-cloud-gcp-starters/pom.xml]: Ah, got your point. I somehow did not wrap my head around the current parent/aggregator structure in `spring-cloud-gcp-starters/pom.xml`. Made some updates, now:

`spring-cloud-previews/pom.xml` is the aggregator pom that will have all generated modules listed. 
- it is used as part of the generation process, to run formatter on all generated modules. This is to get rid of un-used classes in imports, which gapic-generator does not have the capability to do itself. 
- added to `spring-cloud-gcp/pom` `default` profile, to include these modules in build from project root. (when ready to release, add to `release` profile then.)

`spring-cloud-gcp-starters/pom.xml` only serves as parent pom for these modules, modules will not be added to it.

You can see chagnes [here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1394/commits/74c96dc7305e416a670d9bfdcb6e8839188c9d99). What do you think?

Comment (meltsufin) [reply spring-cloud-gcp-starters/pom.xml]: Looks good.

**Review by meltsufin — APPROVED** [review]

File: spring-cloud-generator/generate-one.sh [hunk 3]
```diff
@@ -73,40 +73,46 @@ bazel build //$googleapis_folder:"$client_lib_name"_java_gapic_spring
 cd -
 
 ## copy spring code to outside
-mkdir -p ../generated
-cp googleapis/bazel-bin/$googleapis_folder/"$client_lib_name"_java_gapic_spring-spring.srcjar ../generated
+mkdir -p ../spring-cloud-previews
```
Comment (emmileaf) [inline spring-cloud-generator/generate-one.sh:76]: Nit: could we maybe use an exit code to condition these post-process / add to pom steps on the success status of the bazel build call above?

**Review by emmileaf — COMMENTED** [review]
One additional note that doesn't have to be a part of this PR: given the folder name changes, we should also update the github actions IT suppression from earlier (cc/ @diegomarquezp): https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/c636695390223fecd659fab600fabd4f3e746e13/.github/workflows/integrationTests.yaml#L10-L11

File: spring-cloud-generator/generate-one.sh [hunk 4]
```diff
@@ -73,40 +73,45 @@ bazel build //$googleapis_folder:"$client_lib_name"_java_gapic_spring
...
+cp googleapis/bazel-bin/google/cloud/$client_lib_name/v1/"$client_lib_name"_java_gapic_spring-spring.srcjar ../spring-cloud-previews
```
Comment (emmileaf) [inline spring-cloud-generator/generate-one.sh:77]: This line might have gotten changed unintentionally? 
```suggestion
cp googleapis/bazel-bin/$googleapis_folder/"$client_lib_name"_java_gapic_spring-spring.srcjar ../spring-cloud-previews
```

File: spring-cloud-generator/download-repos.sh [hunk 5]
```diff
@@ -9,7 +9,7 @@ git clone https://github.com/googleapis/googleapis.git
 git clone https://github.com/googleapis/gapic-generator-java.git
 # get into gapic and checkout branch to use
 cd gapic-generator-java
-git checkout autoconfig-gen-draft2
+git checkout replace-shared-w-autoconfig
```
[the remark this answers is not in the document]
Comment (zhumin8) [reply spring-cloud-generator/download-repos.sh:12]: Thanks for catching this!! `pom-changes` has been merged in, so `autoconfig-gen-draft2` would be it.

File: spring-cloud-generator/generate-one.sh (hunk 4 above)
Comment (zhumin8) [reply spring-cloud-generator/generate-one.sh:77]: Great catch! this almost slipped in.

Comment (zhumin8) [reply spring-cloud-generator/generate-one.sh]: fixed in [update](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1394/commits/035e302400e813a8db39eae46e7a0eb13fd92ede)

File: spring-cloud-generator/generate-one.sh (hunk 3 above)
Comment (zhumin8) [reply spring-cloud-generator/generate-one.sh:76]: Added exit on error option to address this for now ([commit](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1394/commits/6af2bdf772b6bcd8df56368ec72915d07a1fdddb)). 
I would keep track of a separate task to improve error handling overall.

**Pull request merged**